examples/sst2/model.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""LSTM classifier model for SST-2."""
+
+import functools
+from typing import Any, Callable, Dict, Text
+
+import flax
+from flax import nn
+import jax
+import jax.numpy as jnp
+from jax import lax
+
+import numpy as np
+
+# pylint: disable=arguments-differ,too-many-arguments
+
+
+@functools.partial(jax.jit, static_argnums=(0, 1, 2, 3))
+def create_model(seed: int, batch_size: int, max_len: int,
+                 model_kwargs: Dict[Text, Any]):
+  """Instantiates a new model."""
+  model_def = TextClassifier.partial(train=False, **model_kwargs)
+  _, initial_params = model_def.init_by_shape(
+      jax.random.PRNGKey(seed),
+      [((batch_size, max_len), jnp.int32),
+       ((batch_size,), jnp.int32)])
+  model = nn.Model(model_def, initial_params)
+  return model
+
+
+def word_dropout(inputs: jnp.ndarray, rate: float = None, unk_idx: int = None):
```

Comment:
Very nice functionality! I was wondering about the interface if this could be more similar to other modules, providing train, and not having default None values for rate and unk_idx. Maybe, just adapting train. Up to you of cause.

Reply:
I added "deterministic" like dropout has, so it's now a more standard API.